### PR TITLE
Add GetDatasetLocks use case

### LINKED ISSUE #95: Create use case for obtaining Dataset locks
**Overview of the Feature Request**

We need to create a new use case for obtaining Dataset locks, as they are required for the SPA.

These changes depend on the existing endpoint for locks: https://github.com/IQSS/dataverse/blob/develop/src/main/java/edu/harvard/iq/dataverse/api/Datasets.java#L2747

**What kind of user is the feature intended for?**

Package consumer

**What inspired the request?**

- SPA needs

**What existing behavior do you want changed?**

None

**Any brand new behavior do you want to add to Dataverse?**

new use case for obtaining Dataset locks

**Any open or closed issues related to this feature request?**

None

### PR DESCRIPTION
**Pull request by GPortas**
## What this PR does / why we need it:

Adds use case for retrieving dataset locks.

This feature is required for the SPA.

## Which issue(s) this PR closes:

- Closes #95 

## Related Dataverse PRs:

Depends on:

- https://github.com/IQSS/dataverse-client-javascript/pull/98
- https://github.com/IQSS/dataverse/pull/10014
- https://github.com/IQSS/dataverse/pull/10023

## Suggestions on how to test this:

- Visual inspection of the code
- Check unit and integration tests execution in GitHub actions

## Is there a release notes update needed for this change?:

N/A

## Additional documentation:

None

### CODE REVIEW

File: src/datasets/domain/models/DatasetLock.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  datasetPersistentId: string;
+  message?: string;
+}
+
+export enum DatasetLockType {
+  WORKFLOW = 'Workflow',
+  IN_REVIEW = 'InReview',
+  DCM_UPLOAD = 'DcmUpload',
+  GLOBUS_UPLOAD = 'GlobusUpload',
+  FINALIZE_PUBLICATION = 'finalizePublication',
```
Comment (ekraffmiller) [inline src/datasets/domain/models/DatasetLock.ts:15]: The values for datasetLockType have aren't capitalized in the frontend, maybe these should be changed to match, and make the mapping easier? ('finalizePublication' already matches). Also, INGEST is not in this list, not sure if it needs to be added.

The enum in the frontend:
```
export enum DatasetLockReason {
  INGEST = 'ingest',
  WORKFLOW = 'workflow',
  IN_REVIEW = 'inReview',
  DCM_UPLOAD = 'dcmUpload',
  GLOBUS_UPLOAD = 'globusUpload',
  FINALIZE_PUBLICATION = 'finalizePublication',

  EDIT_IN_PROGRESS = 'editInProgress',
  FILE_VALIDATION_FAILED = 'fileValidationFailed'
}
```

**Review by ekraffmiller — COMMENTED** [review]
Looks good, I just have a comment about the DatasetLockReason values.

File: src/datasets/domain/models/DatasetLock.ts (hunk 1 above)
Comment (ekraffmiller) [reply src/datasets/domain/models/DatasetLock.ts:15]: To update my previous comment, I can see that these values (capitalization) match the enum in the Daverse API, so I can see leaving them as is, unless we want to fix the capitalization inconsistency in the API?

Comment (GPortas) [reply src/datasets/domain/models/DatasetLock.ts:15]: Thanks, Ellen. I just added the ingest type. It was missing.

Updating it in the API can cause backwards compatibility issues. I would leave it like this for now.

**Comment by GPortas** [comment]
@ekraffmiller We can see the same error that you had yesterday on your localhost, now also in action: https://github.com/IQSS/dataverse-client-javascript/actions/runs/6719463036/job/18261139350

I'm going to link this PR to the issue you created.

**Review by ekraffmiller — APPROVED** [review]
LGTM!

**Pull request merged**